### PR TITLE
Pomodoro feature

### PR DESCRIPTION
**Pull request by junaidshaikh-js**
Pomodoro clock page is added 

- User now click on the task title to go to the Pomodoro clock 
- Clock can start, stop, and can be reset. 
- Notification is shown on each action

### CODE REVIEW

File: src/components/pomoclock/_pomodoro.scss [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  .info-message {
+    text-align: center;
+    font-size: 1.5rem;
+    color: var(--primary-color);
+    position: absolute;
+    top: 10%;
+    right: 1%;
+    background: black;
+    padding: 0.5rem;
```
Comment (saadahmadhere) [inline src/components/pomoclock/_pomodoro.scss:33]: 0.5rem may also be written in absolute units.

**Review by saadahmadhere — COMMENTED** [review]
Astonishing. Nice use of SCSS. Use of hsl for colors added uniqueness to the code.

File: src/components/pomoclock/pomoclock.jsx [hunk 2]
```diff
@@ -1,3 +1,148 @@
...
+
 export function Pomoclock() {
-  return <>Pomoclock</>;
+  const location = useLocation();
+  const { task } = location.state;
+  const { title, description, time } = task;
+  const [seconds, setSeconds] = useState(0);
+  const [infoMessage, setInfoMessage] = useState("");
+  const [setTimeoutId, setSetTimeoutId] = useState(null);
+  const clockInitalState = {
```
Comment (lannisterrr) [inline src/components/pomoclock/pomoclock.jsx:14]: If these states are inter-dependent , then useReducer can be used to manage state.

**Review by lannisterrr — APPROVED** [review]
Good use of useRef and react portals, some suggestions made.

File: src/components/pomoclock/_pomodoro.scss (hunk 1 above)
Comment (junaidshaikh-js) [reply src/components/pomoclock/_pomodoro.scss:33]: Yes, we can do it in multiple ways. I prefer to use the `rem`

File: src/components/pomoclock/pomoclock.jsx (hunk 2 above)
Comment (junaidshaikh-js) [reply src/components/pomoclock/pomoclock.jsx:14]: No, states aren't interdependent.

**Pull request merged**